### PR TITLE
Make adding InputSchemes more User friendly

### LINKED ISSUE #529: Child Scenes should import KeyBindings

### LINKED ISSUE #539: Effectively Map Keys directly to GaInput enums
Effectively Map Keys directly to GaInput enums so that User can directly register InputSchemes without knowing GaInput enums

### PR DESCRIPTION
**Pull request by Bashar-Ahmed**

### CODE REVIEW

File: rootex/framework/scene.cpp [hunk 1]
```diff
@@ -606,9 +606,280 @@ void SceneSettings::drawInputScheme(InputDescription& inputDesc)
...
+			}
+			isEq &= (charIndex == nameSize);
+			currentKey = isEq ? i : currentKey;
+		}
+		ImGui::Combo("", &currentKey, keys, ARRAYSIZE(keys));
+		inputDesc.button = InputManager::GetSingleton()->getMouse()->GetButtonByName(keys[currentKey]);
+	}
+	else if (device == 1)
+	{
+		char* keys[] = {
```
Comment (sin3point14) [inline rootex/framework/scene.cpp]: Store all these arrays in InputManager classes. 
Also prefer using https://github.com/sdslabs/Rootex/blob/dev/rootex/common/types.h#L106 for static arrays
Keep the elements `String` as well, comes handy.

File: rootex/framework/scene.cpp [hunk 2]
```diff
@@ -606,9 +606,280 @@ void SceneSettings::drawInputScheme(InputDescription& inputDesc)
...
+			"mouse_15",
+			"mouse_16",
+			"mouse_17",
+			"mouse_18",
+			"mouse_19",
+			"mouse_20",
+			"mouse_x",
+			"mouse_y"
+		};
+		char name[20] = {};
```
Comment (sin3point14) [inline rootex/framework/scene.cpp]: String

File: rootex/framework/scene.cpp [hunk 3]
```diff
@@ -606,9 +606,280 @@ void SceneSettings::drawInputScheme(InputDescription& inputDesc)
 	ImGui::SameLine();
 	int button = inputDesc.button;
 	ImGui::SetNextItemWidth(ImGui::GetContentRegionAvailWidth() / 3.0f);
-	if (ImGui::InputInt("##button", &button))
-	{
-		inputDesc.button = (DeviceButtonID)button;
+	if (device == 0)
```
Comment (sin3point14) [inline rootex/framework/scene.cpp]: enums https://github.com/sdslabs/Rootex/blob/dev/rootex/core/input/input_manager.h#L19

File: rootex/framework/scene.cpp (hunk 1 above)
Comment (sin3point14) [reply rootex/framework/scene.cpp]: can we construct this array dynamically by looking and calling `GetButtonName`?
creating an array here seems very risky

File: rootex/framework/scene.cpp [hunk 4]
```diff
@@ -606,9 +606,280 @@ void SceneSettings::drawInputScheme(InputDescription& inputDesc)
...
+			int charIndex = 0;
+			while (keys[i][charIndex])
+			{
+				isEq &= (keys[i][charIndex] == name[charIndex]);
+				charIndex++;
+			}
+			isEq &= (charIndex == nameSize);
+			currentKey = isEq ? i : currentKey;
+		}
+		ImGui::Combo("", &currentKey, keys, ARRAYSIZE(keys));
```
Comment (sin3point14) [inline rootex/framework/scene.cpp]: An empty first argument may give us collision issues, if it needs to be empty, add make it something like: `"##Keyboard Input"

File: rootex/core/input/input_manager.cpp [hunk 5]
```diff
@@ -185,6 +185,236 @@ Vector2 InputManager::getMousePosition()
...
+		"colon",
+		"exclam",
+		"brace_left",
+		"brace_right",
+		"sys_rq"
+	};
+	return keyboardKeys;
+}
+
+Array<String, 20> InputManager::getPadButtonName()
```
Comment (angad-k) [inline rootex/core/input/input_manager.cpp]: maybe rename this to `getPadButtonNames`, similar for all others

**Review by angad-k — CHANGES_REQUESTED** [review]
Rename those functions, lgtm otherwise

File: rootex/core/input/input_manager.cpp [hunk 6]
```diff
@@ -185,6 +185,236 @@ Vector2 InputManager::getMousePosition()
 	return { getMouse()->GetFloat(MouseButton::MouseAxisX), getMouse()->GetFloat(MouseButton::MouseAxisY) };
 }
 
+Array<String, 23> InputManager::getMouseButtonName()
+{
+	Array<String, 23> mouseKeys = {
```
Comment (twaritwaikar) [inline rootex/core/input/input_manager.cpp]: Making these arrays static will give you a free perf boost

**Review by angad-k — APPROVED** [review]

**Pull request merged**